### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by mattcollier**
Requests made to a namespace that does not have `meta.status === 'active'` are failed straight away.  As discussed, some background process will be responsible for tagging suspended namespaces.

Applications like kms and edv setup an Express.js `Router`.  Follows the pattern set forth in ledger-agent plugins.  Except that applications as implemented here are not namespace specific.  All namespaces have the same applications available which suits our current use case.

```
config.namespace.applications.kms = new Router();
```
The `req` passed in the application route will have information about the namespace in `req.namespace` including the zcap to be invoked to authorize a `create` operation, and/or to record a successful operation.  Possibly two different zcaps?

@dlongley want to ensure that the general direction is proper before refactoring bedrock-kms-http to work within the namespace framework.

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*!
+ * Copyright (c) 2020-2021 Digital Bazaar, Inc. All rights reserved.
+ */
+
+// FIXME: is this right? Why is the host coming from the request instead
+// of the bedrock config?
```
Comment (dlongley) [inline lib/helpers.js:6]: Seems like it should come from the bedrock config. We'd only need to take it from the request if we were going to support multiple hosts on a server -- and if we could be guaranteed to trust that host. I don't think we have that sort of use case anymore?

File: lib/http.js [hunk 2]
```diff
@@ -3,12 +3,54 @@
...
+
+  app.use(routes.application, asyncHandler(async (req, res, next) => {
+    const {params: {namespaceId, application}} = req;
+    const id = getNamespaceId({req, localId: namespaceId, routes});
+
+    const {config: namespaceConfig, meta} = await namespace.get({id});
+
+    if(meta.status !== 'active') {
+      throw new BedrockError(
+        'The namespace is not active.', 'InvalidStateError', {
```
Comment (dlongley) [inline lib/http.js:36]: I don't think being inactive is an invalid state, it just means using the namespace is not allowed.

```suggestion
        'The namespace is not active.', 'NotAllowedError', {
```

**Review by dlongley — COMMENTED** [review]
AFAICT, this seems like an ok direction so far.

**Comment by mattcollier** [comment]
TY, closing PR for now.

**Pull request closed without merging**